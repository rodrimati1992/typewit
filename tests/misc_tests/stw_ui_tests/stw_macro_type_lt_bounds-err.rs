use typewit::MakeTypeWitness;

use core::fmt::Debug;


typewit::simple_type_witness!{
    enum TypeParams['a, T, U:, V: 'a, W: 'a + 'static, X: Sized, Y: 'a + Sized] {
        U8 = (&'a (), T, U, V, W, X, Y) 
    }
}

typewit::simple_type_witness!{
    enum TypeParamsAngBrack<'a, T, U:, V: 'a, W: 'a + 'static, X: Sized, Y: 'a + Sized> {
        U8 = (&'a (), T, U, V, W, X, Y) 
    }
}

fn _foo<'a, T, U, V, W, X, Y>() {
    let _: TypeParams<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
}

fn _bar<'a, T, U, V, W, X, Y>() {
    let _: TypeParamsAngBrack<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
}

fn main(){}