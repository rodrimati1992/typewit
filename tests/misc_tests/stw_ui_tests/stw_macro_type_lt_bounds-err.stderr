warning: unused import: `core::fmt::Debug`
 --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:3:5
  |
3 | use core::fmt::Debug;
  |     ^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0309]: the parameter type `V` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:19:12
   |
19 |     let _: TypeParams<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ...so that the type `V` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
18 | fn _foo<'a, T, U, V: 'a, W, X, Y>() {
   |                    ++++

error[E0309]: the parameter type `W` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:19:12
   |
19 |     let _: TypeParams<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ...so that the type `W` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
18 | fn _foo<'a, T, U, V, W: 'a, X, Y>() {
   |                       ++++

error[E0310]: the parameter type `W` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:19:12
   |
19 |     let _: TypeParams<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ...so that the type `W` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
18 | fn _foo<'a, T, U, V, W: 'static, X, Y>() {
   |                       +++++++++

error[E0309]: the parameter type `Y` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:19:12
   |
19 |     let _: TypeParams<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ...so that the type `Y` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
18 | fn _foo<'a, T, U, V, W, X, Y: 'a>() {
   |                             ++++

error[E0309]: the parameter type `V` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:19:50
   |
19 |     let _: TypeParams<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |                                                  ^^^^^^^^^^^^^^^^^^^^^ ...so that the type `V` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
18 | fn _foo<'a, T, U, V: 'a, W, X, Y>() {
   |                    ++++

error[E0309]: the parameter type `W` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:19:50
   |
19 |     let _: TypeParams<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |                                                  ^^^^^^^^^^^^^^^^^^^^^ ...so that the type `W` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
18 | fn _foo<'a, T, U, V, W: 'a, X, Y>() {
   |                       ++++

error[E0310]: the parameter type `W` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:19:50
   |
19 |     let _: TypeParams<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |                                                  ^^^^^^^^^^^^^^^^^^^^^ ...so that the type `W` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
18 | fn _foo<'a, T, U, V, W: 'static, X, Y>() {
   |                       +++++++++

error[E0309]: the parameter type `Y` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:19:50
   |
19 |     let _: TypeParams<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |                                                  ^^^^^^^^^^^^^^^^^^^^^ ...so that the type `Y` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
18 | fn _foo<'a, T, U, V, W, X, Y: 'a>() {
   |                             ++++

error[E0309]: the parameter type `V` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:23:12
   |
23 |     let _: TypeParamsAngBrack<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ...so that the type `V` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
22 | fn _bar<'a, T, U, V: 'a, W, X, Y>() {
   |                    ++++

error[E0309]: the parameter type `W` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:23:12
   |
23 |     let _: TypeParamsAngBrack<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ...so that the type `W` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
22 | fn _bar<'a, T, U, V, W: 'a, X, Y>() {
   |                       ++++

error[E0310]: the parameter type `W` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:23:12
   |
23 |     let _: TypeParamsAngBrack<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ...so that the type `W` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
22 | fn _bar<'a, T, U, V, W: 'static, X, Y>() {
   |                       +++++++++

error[E0309]: the parameter type `Y` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:23:12
   |
23 |     let _: TypeParamsAngBrack<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ...so that the type `Y` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
22 | fn _bar<'a, T, U, V, W, X, Y: 'a>() {
   |                             ++++

error[E0309]: the parameter type `V` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:23:58
   |
23 |     let _: TypeParamsAngBrack<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |                                                          ^^^^^^^^^^^^^^^^^^^^^ ...so that the type `V` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
22 | fn _bar<'a, T, U, V: 'a, W, X, Y>() {
   |                    ++++

error[E0309]: the parameter type `W` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:23:58
   |
23 |     let _: TypeParamsAngBrack<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |                                                          ^^^^^^^^^^^^^^^^^^^^^ ...so that the type `W` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
22 | fn _bar<'a, T, U, V, W: 'a, X, Y>() {
   |                       ++++

error[E0310]: the parameter type `W` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:23:58
   |
23 |     let _: TypeParamsAngBrack<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |                                                          ^^^^^^^^^^^^^^^^^^^^^ ...so that the type `W` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
22 | fn _bar<'a, T, U, V, W: 'static, X, Y>() {
   |                       +++++++++

error[E0309]: the parameter type `Y` may not live long enough
  --> tests/misc_tests/stw_ui_tests/stw_macro_type_lt_bounds-err.rs:23:58
   |
23 |     let _: TypeParamsAngBrack<'a, T, U, V, W, X, Y, _> = MakeTypeWitness::MAKE;
   |                                                          ^^^^^^^^^^^^^^^^^^^^^ ...so that the type `Y` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
22 | fn _bar<'a, T, U, V, W, X, Y: 'a>() {
   |                             ++++
